frontend/simple/js/state.js

```diff
@@ -96,12 +141,20 @@ const actions = {
   ) {
     await dispatch('saveSettings')
     await db.clearCurrentUser()
-    commit('logout')
     for (let key of Object.keys(state.contracts)) {
       await backend.unsubscribe(key)
     }
+    commit('logout')
     Vue.events.$emit('logout')
   },
+  deleteMail ({dispatch, commit}, hash) {
+    commit('deleteMail', hash)
+    debouncedSave(dispatch)
+  },
+  readMail ({dispatch, commit}, hash) {
+    commit('readMail', hash)
+    debouncedSave(dispatch)
+  },
```

Comment:
This looks like unnecessary code duplication to me... Is there a more generic, DRY-respecting way to do this?